source/windows/clock.c

```diff
@@ -51,10 +51,15 @@ int aws_high_res_clock_get_ticks(uint64_t *timestamp) {
     LARGE_INTEGER ticks, frequency;
     /* QPC runs on sub-microsecond precision, convert to nanoseconds */
     if (QueryPerformanceFrequency(&frequency) && QueryPerformanceCounter(&ticks)) {
-        *timestamp = aws_timestamp_convert((uint64_t)ticks.QuadPart, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_MICROS, NULL) /
-                     (uint64_t)frequency.QuadPart;
+        uint64_t u64_ticks = ticks.QuadPart;
+        uint64_t u64_frequency = frequency.QuadPart;
+
+        uint64_t seconds = u64_ticks / u64_frequency;
+        uint64_t ticks_remainder = u64_ticks - seconds * u64_frequency;
+
+        *timestamp = aws_timestamp_convert(seconds, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL);
+        *timestamp += aws_timestamp_convert_u64(ticks_remainder, u64_frequency, AWS_TIMESTAMP_NANOS, NULL);
```

Comment:
you don't need to do all this other math. you can just do:
`*timestamp = aws_timestamp_convert_u64(u64_tick, u64_frequency, AWS_TIMESTAMP_NANOS, NULL);`

Reply:
Magic